components/timezone/src/iana_ids.rs

```diff
@@ -0,0 +1,243 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::error::TimeZoneError;
+use crate::provider::names::*;
+use crate::TimeZoneBcp47Id;
+use icu_provider::prelude::*;
+
+/// A mapper from IANA time zone identifiers to BCP-47 time zone identifiers.
+///
+/// # Examples
+///
+/// ```
+/// use icu::timezone::IanaToBcp47Mapper;
+///
+/// let mapper = IanaToBcp47Mapper::new();
+/// let mapper_borrowed = mapper.as_borrowed();
+///
+/// // The IANA zone "Australia/Melbourne" is the BCP-47 zone "aumel"
+/// assert_eq!(
+///     mapper_borrowed.get("Australia/Melbourne"),
+///     Some("aumel".parse().unwrap())
+/// );
+///
+/// // Lookup is ASCII-case insensitive
+/// assert_eq!(
+///     mapper_borrowed.get("australia/melbourne"),
+///     Some("aumel".parse().unwrap())
+/// );
+///
+/// // The IANA zone "Australia/Victoria" is an alias
+/// assert_eq!(
+///     mapper_borrowed.get("Australia/Victoria"),
+///     Some("aumel".parse().unwrap())
+/// );
+/// ```
+#[derive(Debug)]
+pub struct IanaToBcp47Mapper {
+    data: DataPayload<IanaToBcp47MapV1Marker>,
+}
+
+impl IanaToBcp47Mapper {
+    /// Creates a new [`IanaToBcp47Mapper`].
+    ///
+    /// See [`IanaToBcp47Mapper`] for an example.
+    ///
```

Comment:
Kind of obvious

```suggestion

```

Reply:
We already commonly link back to struct- or module-level docs for terminal functions where people may look for examples. We have a style guide requirement that all functions have examples and I consider this a substitute for that when the example would be fully duplicated.